lib/streamlit/deprecation_util.py

```diff
@@ -21,10 +21,15 @@
 TObj = TypeVar("TObj", bound=object)
 
 
-def _show_deprecated_name_warning_in_browser(
+def show_deprecation_warning(message: str) -> None:
+    """Show a deprecation warning message in the browser."""
+    streamlit.warning(message)
```

Comment:
`@st.cache` required custom deprecation logic, so I exposed this function directly.